### ISSUE TITLE
sonarjs/cognitive-complexity doesn't ignore functional components written as arrow function 

### ISSUE THREAD

**Issue by neotrow** [SonarSource/eslint-plugin-sonarjs#422]
**I want to report a bug.**

I saw that react functional components should be ignored: https://github.com/SonarSource/SonarJS/issues/2238

But if the functional component is written as arrow function it's not ignored.

This will not be ignored and complain about cognitiv complexity:

```typescript
const EmployeeDetailsSection: FC<EmployeeDetailsSectionProps> = (props) => {
  const t = useT();
  const { locale } = useRouter();

  return (
    <div>
       ...
   </div>
  );
}
```

But this will be ignored:

```typescript
function EmployeeDetailsSection(props) {
  const t = useT();
  const { locale } = useRouter();

  return (
    <div>
       ...
   </div>
  );
}
```

**eslint-plugin-sonarjs version:** 0.21.0
**eslint version:** 8.26.0
**Node.js version:** 20

**Rule key:** sonarjs/cognitive-complexity

**Comment by btynflvc63** [comment]
This will not be ignored and complain about cognitiv complexity:

`const EmployeeDetailsSection = React.forwardRef((props, ref) => {
  const t = useT();
  const { locale } = useRouter();

  return (
    <div>
       ...
   </div>
  );
})
`

**Comment by sebheron** [comment]
This is still an issue. Is this being worked on?

**Comment by Wohops** [comment]
This issue has been migrated to Jira. [ESLINTJS-27](https://sonarsource.atlassian.net/browse/ESLINTJS-27)
